mlir/lib/Conversion/GPUToLLVMSPV/GPUToLLVMSPV.cpp

```diff
@@ -325,16 +359,45 @@ struct GPUToLLVMSPVConversionPass final
     LLVMTypeConverter converter(context, options);
     LLVMConversionTarget target(*context);
 
+    if (forceOpenclAddressSpaces) {
+      MemorySpaceToOpenCLMemorySpaceConverter converter;
+      AttrTypeReplacer replacer;
+      replacer.addReplacement([&converter](BaseMemRefType origType)
+                                  -> std::optional<BaseMemRefType> {
+        return converter.convertType<BaseMemRefType>(origType);
+      });
+
+      replacer.recursivelyReplaceElementsIn(getOperation(),
+                                            /*replaceAttrs=*/true,
+                                            /*replaceLocs=*/false,
+                                            /*replaceTypes=*/true);
+    }
+
     target.addIllegalOp<gpu::BarrierOp, gpu::BlockDimOp, gpu::BlockIdOp,
                         gpu::GPUFuncOp, gpu::GlobalIdOp, gpu::GridDimOp,
                         gpu::ReturnOp, gpu::ShuffleOp, gpu::ThreadIdOp>();
 
     populateGpuToLLVMSPVConversionPatterns(converter, patterns);
+    populateFuncToLLVMConversionPatterns(converter, patterns);
+    populateFinalizeMemRefToLLVMConversionPatterns(converter, patterns);
```

Comment:
I'd avoid this as this is supposed to be an atomic pass. If users wanna have a more complex pass, they're free to use this pass's patterns and build a more complex one.

Reply:
These are to handle the callees (which are `func.func` and hence aren't handled at all). I think the pass would be just incomplete without it since it leaves out memrefs between the caller and the callee, WDYT?

Reply:
Users could create a pass, use patterns exposed by this pass, use more patterns thanks to the convert to LLVM interface and populate memspace conversion with the entrypoint exposed by our pass. I think it's alright to drop this, keep pass simple and let users extend their passes at their will/use more than one pass.